library/ssl_tls.c

```diff
@@ -6459,6 +6639,42 @@ int mbedtls_ssl_config_defaults( mbedtls_ssl_config *conf,
     int ret = MBEDTLS_ERR_ERROR_CORRUPTION_DETECTED;
 #endif
 
+#if defined(MBEDTLS_DEBUG_C) && defined(MBEDTLS_KEY_EXCHANGE_WITH_CERT_ENABLED)
+    if( ssl_array_has_duplicated_entries( ssl_preset_suiteb_sig_algs ) )
+    {
+#if defined(MBEDTLS_PLATFORM_C)
```

Comment:
It seems you should rather add at the beginning of the file something like:
```C
#if !defined(MBEDTLS_PLATFORM_C)                                                
#include <stdio.h>                                                              
#define mbedtls_printf printf                                                   
#endif /* MBEDTLS_PLATFORM_C */
```
See gcm.c for example.